src/pudl/etl.py

```diff
@@ -15,326 +15,187 @@
-        """
-        SELECT DISTINCT strftime('%Y', report_date)
-        AS year
-        FROM plants_eia860
-        ORDER BY year ASC
-        """, pudl_engine).year.astype(int)
-    missing_years = list(set(epacems_years) - set(eia_plant_years))
-    if missing_years:
-        logger.info(
-            f"EPA CEMS years with no EIA plant data: {missing_years} "
-            "Some timezones may be estimated based on plant state."
-        )
-
-    # NOTE: This is a generator for raw dataframes
-    epacems_raw_dfs = pudl.extract.epacems.extract(
-        epacems_years, epacems_states, Datastore(**ds_kwargs))
-
-    # NOTE: This is a generator for transformed dataframes
-    epacems_transformed_dfs = pudl.transform.epacems.transform(
-        epacems_raw_dfs=epacems_raw_dfs,
-        pudl_engine=pudl_engine,
-    )
-
-    logger.info("Processing EPA CEMS data and writing it to Apache Parquet.")
-    if logger.isEnabledFor(logging.INFO):
-        start_time = time.monotonic()
-
-    # run the cems generator dfs through the load step
-    for df in epacems_transformed_dfs:
-        pudl.load.parquet.epacems_to_parquet(
-            df,
-            root_path=Path(pudl_settings["parquet_dir"]) / "epacems",
-        )
-
-    if logger.isEnabledFor(logging.INFO):
-        delta_t = time.strftime("%H:%M:%S", time.gmtime(
-            time.monotonic() - start_time))
-        time_message = f"Processing EPA CEMS took {delta_t}"
-        logger.info(time_message)
-        start_time = time.monotonic()
-
-
-###############################################################################
-# GLUE EXPORT FUNCTIONS
-###############################################################################
-
-def _etl_glue(etl_settings: GlueSettings) -> Dict[str, pd.DataFrame]:
-    """Extract, transform and load CSVs for the Glue tables.
-
-    Args:
-        etl_settings (GlueSettings): Validated ETL parameters required by this data source.
-
-    Returns:
-        dict: A dictionary of :class:`pandas.Dataframe` whose keys are the names
-        of the corresponding database table.
-
+    onfail_success = (commandline_args.keep_cache == 'ONFAIL' and state.is_successful())
+    cache_root = commandline_args.pipeline_cache_path
+
+    if (commandline_args.keep_cache == 'ALWAYS') or not onfail_success or commandline_args.rerun:
+        logger.warning(f'Keep pipeline cache director under {cache_root}')
+        return
+    if (commandline_args.keep_cache == 'NEVER') or onfail_success:
+        if not cache_root.startswith("gs://"):
+            logger.warning(f'Deleting pipeline cache directory under {cache_root}')
+            # TODO(rousik): in order to prevent catastrophic results due to misconfiguration
+            # we should refuse to delete cache_root unless the directory has the expected
+            # run_id form of YYYY-MM-DD-HHMM-uuid4
+            shutil.rmtree(cache_root)
+
+
+def configure_prefect_context(etl_settings, pudl_settings, commandline_args):
```

Comment:
Prefect.context is a global var store that can be accessed during a flow run. Things like etl_settings, pudl_settings, overwrite_ferc1_db and pipeline_cache_path are all stored in prefect.context. Do y’all have any opinions on what should and should not live in this?